### PR TITLE
2022/base styles

### PR DESCRIPTION
**Pull request by zastrow**
This setups the base  styles including colors and typography for the 2022 design.

To test:

1. Run `npm install`
2. Run `npm start`
3. Open [http://localhost:8000/2022](http://localhost:8000/2022) in your browser
4. Review heading styles and body styles against the [Figma styles](https://www.figma.com/file/VcIuWHbkjExLZ8SNQZipAo/2022-DS-Survey?node-id=2207%3A1035)

Check the following styles:
- [x] **Heading Styles**
   - [x] Section Header
   - [x] Header 02
   - [x] Header 03
   - [x] Header 04
- [x] **Body Styles**
   - [x] Body Default
   - [x] Body Default Link
   - [x] Body Large
   - [x] Body Large Regular
   - [x] Body X Large
   - [x] Description

### CODE REVIEW

File: src/scss/2022/settings/_colors.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+$colors: (
+  'white': '58, 30%, 96%',
```
Comment (bengoshow) [inline src/scss/2022/settings/_colors.scss]: Nailed it. 👏

File: src/sections/2022/demo.mdx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<p>Heading Styles</p>
+<h1 class="cmp-type-heading-1">Section Heading</h1>
+<h2 class="cmp-type-heading-2">Heading 02</h2>
+<h3 class="cmp-type-heading-3">Heading 03</h3>
+<h4 class="cmp-type-heading-3">Heading 04</h4>
```
Comment (bengoshow) [inline src/sections/2022/demo.mdx:5]: I think this class name should be changed, right?
```suggestion
<h4 class="cmp-type-heading-4">Heading 04</h4>
```

File: src/scss/2022/settings/_font-size.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+$sizes: (
+  'heading-1': 4.5rem,
+  'heading-2': 3.5rem,
+  'heading-3': 2.5rem,
+  'heading-4': 2rem,
+  'section-description': 1.75rem,
+  'body-xlarge': 1.5rem,
+  'body-large': 1.25rem,
+  'body-default': 1.125rem
```
Comment (bengoshow) [inline src/scss/2022/settings/_font-size.scss:9]: The Figma design calls for 17px, which translates to 1.0625rem
```suggestion
  'body-default': 1.0625rem
```

**Review by bengoshow — COMMENTED** [review]
Left a few comments, @zastrow -- thanks!

File: src/scss/2022/settings/_font-size.scss (hunk 3 above)
Comment (zastrow) [reply src/scss/2022/settings/_font-size.scss:9]: Well, you caught me trying to slide past my irrational dislike for odd numbers and decimal points larger than 3. 😉

Comment (bengoshow) [reply src/scss/2022/settings/_font-size.scss]: Hey, I agree! I'm on your side here - don't shoot the messenger! 😉

**Review by bengoshow — APPROVED** [review]
Looks great, Z! 
very-nice

File: src/scss/2022/tools/_typography.scss [hunk 4]
```diff
@@ -0,0 +1,12 @@
+@mixin style($size, $weight: 200) {
+  @if $size == 'heading-1'
+   or $size == 'heading-2'
+   or $size == 'heading-3'
+   or $size == 'heading-4'
+   or $size == 'section-description' {
+    font-family: var(--font-display);
+  }
```
Comment (QuinnrElder) [inline src/scss/2022/tools/_typography.scss:8]: Everything LGTM! I can compile and run things locally on my M1 with @bengoshow changes [here](https://github.com/sparkbox/designsystemssurvey/pull/336)

That said when I was looking through the Figma design and what was computed in localhost. I saw that the line-heights were not matching up in the H1, H2, H4, and one or two of the Body typographies (off by a few ```px```). Also, the font-size was off by a few ```px``` too. 

Not sure if that is worth holding up the PR. Thoughts on this?
Side note : Im new to working within design systems

Comment (zastrow) [reply src/scss/2022/tools/_typography.scss:8]: @QuinnrElder that is a good catch, I like your attention to detail. One of the things you will hear me quote often is Dan Mall’s statement about design called “deciding in the browser.” Typography changes occur once the content starts to show up in the browser, so `line-height` especially, I keep a few things broad until further along in development.

Comment (QuinnrElder) [reply src/scss/2022/tools/_typography.scss:8]: @zastrow ~ thank you for taking the time to go over that with me! You rock!

**Pull request merged**